README.md

```diff
@@ -673,7 +672,22 @@ We alway welcome your contributions :clap:
 
 Bugs, feature requests and comments are more than welcome in the [issues].
 
-### Feedback
+## Release Process
+
+There is a `release` target within the Makefile that wraps up the steps to release a new version.
```

Comment:
Nit: To follow up #122, I'd suggest to add a line break in .md file anywhere at around 80 char mark.